providers/flipt/src/main/java/dev/openfeature/contrib/providers/flipt/FliptProvider.java

```diff
@@ -0,0 +1,240 @@
+                .reason(valueProviderEvaluation.getReason())
+                .flagMetadata(valueProviderEvaluation.getFlagMetadata())
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue, EvaluationContext ctx) {
+        ProviderEvaluation<Value> valueProviderEvaluation = getObjectEvaluation(key, new Value(defaultValue), ctx);
+        Integer value = getIntegerValue(valueProviderEvaluation, defaultValue);
+        return ProviderEvaluation.<Integer>builder()
+            .value(value)
+            .variant(valueProviderEvaluation.getVariant())
+            .errorCode(valueProviderEvaluation.getErrorCode())
+            .reason(valueProviderEvaluation.getReason())
+            .flagMetadata(valueProviderEvaluation.getFlagMetadata())
+            .build();
+    }
+
+    private static Integer getIntegerValue(ProviderEvaluation<Value> valueProviderEvaluation, Integer defaultValue) {
+        String valueStr = valueProviderEvaluation.getValue().asObject().toString();
+        try {
+            return Integer.parseInt(valueStr);
+        } catch (NumberFormatException ex) {
+            return defaultValue;
+        }
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue, EvaluationContext ctx) {
+        ProviderEvaluation<Value> valueProviderEvaluation = getObjectEvaluation(key, new Value(defaultValue), ctx);
+        Double value = getDoubleValue(valueProviderEvaluation, defaultValue);
+        return ProviderEvaluation.<Double>builder()
+            .value(value)
+            .variant(valueProviderEvaluation.getVariant())
+            .errorCode(valueProviderEvaluation.getErrorCode())
+            .reason(valueProviderEvaluation.getReason())
+            .flagMetadata(valueProviderEvaluation.getFlagMetadata())
+            .build();
+    }
+
+    private static Double getDoubleValue(ProviderEvaluation<Value> valueProviderEvaluation, Double defaultValue) {
+        String valueStr = valueProviderEvaluation.getValue().asObject().toString();
+        try {
+            return Double.parseDouble(valueStr);
+        } catch (NumberFormatException ex) {
+            return defaultValue;
+        }
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        Map<String, String> contextMap = buildContextMap(ctx);
+        EvaluationRequest request = EvaluationRequest.builder().namespaceKey(fliptProviderConfig.getNamespace())
+            .flagKey(key).entityId(ctx.getTargetingKey()).context(contextMap).build();
```

Comment:
`entityId(ctx.getTargetingKey())`

Is there any issue if `ctx.getTargetingKey()` is null here?

Reply:
I don't see an issue, it is working even when _ctx.getTargetingKey()_ is null, as entity Id is null. Confirmed via running a unit test via real Flipt instance.